### ISSUE TITLE
[arm/Windows] Build is broken by PR #14228

### ISSUE THREAD

**Issue by 4creators** [dotnet/coreclr#14514]
PR #14228 introduced two bugs into build.cmd script which prevent successful cross build for Windows arm target.

First issue is caused by configuration of vswhere script which detects only RTM installations of Visual Studio.  In case only preview installation would have arm VC tools installed building with build.cmd will be impossible.

Second error masked by the first one is that execution of  VsDevCmd.bat script will change working directory of build.cmd script. If this is not corrected build fails with "Command line too long" or "Path too long" errors.

**Comment by 4creators** [comment]
Build output with current script run on machine with both VS 2017 RTM and Preview installed by using Preview developer command prompt is shown below. From this run it is visible that only VS RTM is selected and current working directory of the script is changed to default source directory set in VS which is `C:\Users\developer\Source` while repo root is at `e:\src\ms\dotnet\coreclr`.

```
e:\src\ms\dotnet\coreclr>set __ThisScriptFull="e:\src\ms\dotnet\coreclr\build.cmd" 
e:\src\ms\dotnet\coreclr>if defined VisualStudioVersion goto :Run 
e:\src\ms\dotnet\coreclr>set _VSWHERE="C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" 
e:\src\ms\dotnet\coreclr>if exist "C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" (for /F "usebackq tokens=*" %i in (`"C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -latest -property installationPath`) do set _VSCOMNTOOLS=%i\Common7\Tools ) 

e:\src\ms\dotnet\coreclr>set _VSCOMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools 
e:\src\ms\dotnet\coreclr>if not exist "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools" set _VSCOMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools\ 
e:\src\ms\dotnet\coreclr>if not exist "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools" (
echo Error: Visual Studio 2015 or 2017 required.  
 echo        Please see https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/developer-guide.md for build instructions.  
 exit /b 1 
) 

e:\src\ms\dotnet\coreclr>call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools\VsDevCmd.bat" 
**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.4.0
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************
BUILD: Commencing CoreCLR Repo build
BUILD: Checking prerequisites
BUILD: Number of available CPU cores 4
BUILD: Using environment: "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools\\VsDevCmd.bat"
**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.4.0
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************

C:\Users\developer\Source>setlocal
```

**Comment by 4creators** [comment]
Closing - fixed with #14515
